src/PlusDataCollection/Andor/vtkPlusAndorCamera.cxx

```diff
@@ -0,0 +1,956 @@
+{
+  unsigned rawFrameSize = frameSize[0] * frameSize[1];
+  rawFrame.resize(rawFrameSize, 0);
+
+  checkStatus(::SetExposureTime(exposure), "SetExposureTime");
+  checkStatus(::SetShutter(1, shutterMode, 0, 0), "SetShutter");
+
+  int hbin = binning > 0 ? binning : this->HorizontalBins;
+  int vbin = binning > 0 ? binning : this->VerticalBins;
+  checkStatus(::SetImage(hbin, vbin, 1, 1024, 1, 1024), "Binning");
+
+  int vsInd = vsSpeed >= 0 ? vsSpeed : this->VSSpeed;
+  checkStatus(::SetVSSpeed(vsInd), "SetVSSpeed");
+
+  int hsInd = hsSpeed >= 0 ? hsSpeed : this->HSSpeed[1];
+  checkStatus(::SetHSSpeed(this->HSSpeed[0], hsInd), "SetHSSpeed");
+
+  checkStatus(StartAcquisition(), "StartAcquisition");
+  unsigned result = checkStatus(WaitForAcquisition(), "WaitForAcquisition");
+  if(result == DRV_NO_NEW_DATA)   // Log a more specific log message for WaitForAcquisition
+  {
+    LOG_ERROR("Non-Acquisition Event occurred.(e.g. CancelWait() called)");
+  }
+  this->currentTime = vtkIGSIOAccurateTimer::GetSystemTime();
+
+  // iKon-M 934 has 16-bit digitization
+  // https://andor.oxinst.com/assets/uploads/products/andor/documents/andor-ikon-m-934-specifications.pdf
+  // so we choose 16-bit unsigned
+  // GetMostRecentImage() is 32 bit signed variant
+  checkStatus(GetMostRecentImage16(&rawFrame[0], rawFrameSize), "GetMostRecentImage16");
+
+  return PLUS_SUCCESS;
+}
+
+// ----------------------------------------------------------------------------
+void vtkPlusAndorCamera::AddFrameToDataSource(DataSourceArray& ds)
+{
+  for(unsigned i = 0; i < ds.size(); i++)
+  {
+    if(ds[i]->AddItem(&rawFrame[0],
+                      US_IMG_ORIENT_MF,
+                      frameSize, VTK_UNSIGNED_SHORT,
+                      1, US_IMG_BRIGHTNESS, 0,
+                      this->FrameNumber,
+                      currentTime,
+                      UNDEFINED_TIMESTAMP,
+                      nullptr) != PLUS_SUCCESS)
+    {
+      LOG_WARNING("Error adding item to AndorCamera video source " << ds[i]->GetSourceId());
+    }
+    else
+    {
+      LOG_INFO("Success adding item to AndorCamera video source " << ds[i]->GetSourceId());
+    }
+  }
+}
+
+
+// ----------------------------------------------------------------------------
+void vtkPlusAndorCamera::ApplyFrameCorrections()
+{
+  cv::Mat cvIMG(frameSize[0], frameSize[1], CV_16UC1, &rawFrame[0]); // uses rawFrame as buffer
+  cv::Mat floatImage;
+  cvIMG.convertTo(floatImage, CV_32FC1);
+  cv::Mat result;
+
+  cv::subtract(floatImage, cvBiasCorrection, floatImage, cv::noArray(), CV_32FC1);
+  LOG_INFO("Applied constant bias correction");
+
+  // Divide the image by the 32-bit floating point correction image
+  cv::divide(floatImage, cvFlatCorrection, floatImage, 1, CV_32FC1);
+  LOG_INFO("Applied multiplicative flat correction");
+
+  // OpenCV's lens distortion correction
+  cv::undistort(floatImage, result, cvCameraIntrinsics, cvDistanceCoefficients);

```

Comment:
Based on recent conversations, the lens distortion correction should be applied before flat correction.